File: src/001.Array/442.FindAllDuplicatesInArray/442.数组中重复的数据.go
/*
 * @lc app=leetcode.cn id=442 lang=golang
 *
 * [442] 数组中重复的数据
 */

// @lc code=start
func findDuplicates(nums []int) []int {
	targets := make([]int, len(nums))
	res := make([]int, 0)

	for i := 0; i < len(nums); i++ {
		j := nums[i] - 1
		if targets[j] < 1 {
			targets[j] = 1
			continue
		}
		res = append(res, nums[i])
	}

	return res
}

// @lc code=end

File: src/001.Array/442.FindAllDuplicatesInArray/442.数组中重复的数据.ts
/*
 * @lc app=leetcode.cn id=442 lang=typescript
 *
 * [442] 数组中重复的数据
 */

// @lc code=start
function findDuplicates(nums: number[]): number[] {
	let res: number[] = [];
	for (const num of nums) {
		const absNum = Math.abs(num);
		if (nums[absNum - 1] < 0) {
			res.push(absNum);
		} else {
			nums[absNum - 1] = -1 * nums[absNum - 1];
		}
	}
    return res;
}
// @lc code=end
